ios/ElectrodeApiImpl/APIImpls/Navigation/ENNavigationDelegate.swift

```diff
@@ -222,18 +222,28 @@ import UIKit
         }
         button.isEnabled = !(navigationButton.disabled ?? false)
         button.stringTag = navigationButton.id
+        button.currViewController = vc
         return button
     }
 
+    private func emitERNEvent(sender: ENBarButtonItem, stringTag: String) {
+        var viewId = "NOT_SET"
+        if let vc = sender.currViewController as? MiniAppNavViewController, let viewIdentifier = vc.viewIdentifier {
+            viewId = viewIdentifier
+        }
+        ENNavigationAPIImpl.shared.navigationAPI.events.emitEventOnNavButtonClick(buttonId: stringTag)
```

Comment:
Can we also add a deprecated note here for this event.

Reply:
👍 added